File: range_extraction.py
def solution(args):
    i = 0
    while i < len(args):
        j = i 
        while j + 1 < len(args) and args[j] + 1 == args[j + 1]:
            j += 1
        if j != i and j - i > 1:
            args[i:j+1] = [f"{args[i]}-{args[j]}"]
        i += 1
    return ",".join(map(str, args))

File: Javascript/range_extraction.js
function solution(numbers){
    if (!numbers || numbers.length === 0) {
      return "";
    }
    
    let result = [];
    let start = numbers[0];
    let end = numbers[0];
    
    for (let i = 1; i < numbers.length; i++) {
      if (numbers[i] === end + 1) {
        end = numbers[i];
      } else {
        if (end - start >= 2) {
          result.push(`${start}-${end}`);
        } else {
          for (let j = start ; j <= end; j++) {
            result.push(j.toString());
          }
        }
        start = end = numbers[i];
      }
    }
    
    if ( end - start >= 2) {
      result.push(`${start}-${end}`);
    } else {
      for (let j = start; j <= end; j++) {
        result.push(j.toString());
      }
    }
    
    return result.join(",");
    // TODO: complete solution 
  }